doc/source/serve/advanced-guides/multi-app-container.md

```diff
@@ -0,0 +1,175 @@
+
+In this example, we are going to walk through deploying two applications in separate containers: a whisper model and a Resnet50 image classification model. 
+
+First, let's set up the docker images with the required dependencies. 
+
+:::note
+The Ray version and Python version in the container should match those of the host environment exactly. Note that for Python, the versions must match down to the patch number.
+:::
+
+In `whisper.Dockerfile`:
+* We are running the Whisper model on GPU, so the base image for the Whisper Dockerfile is the latest Ray GPU image, `rayproject/ray:latest-py38-gpu`.
+* We install the package `faster_whisper`, which is a dependency for the whisper model.
+* We then download the source code for the whisper application into `whisper_example.py`.
+Similarly, `resnet.Dockerfile` uses the appropriate base image, installs necessary dependencies, and downloads the resnet model source code.
+Save the following to files named `whisper.Dockerfile` and `resnet.Dockerfile`.
+
+::::{tab-set}
+:::{tab-item} whisper.Dockerfile
+```dockerfile
+FROM rayproject/ray:latest-py38-gpu
+
+RUN pip install faster_whisper==0.10.0
+RUN sudo apt-get update && sudo apt-get install curl -y
+
+RUN curl -O https://raw.githubusercontent.com/ray-project/ray/master/doc/source/serve/doc_code/whisper_example.py
+```
+:::
+:::{tab-item} resnet.Dockerfile
+```dockerfile
+FROM rayproject/ray:latest-py38-cpu
+
+RUN torch==2.0.1 torchvision==0.15.2
+RUN sudo apt-get update && sudo apt-get install curl -y
+
+RUN curl -O https://raw.githubusercontent.com/ray-project/ray/master/doc/source/serve/doc_code/resnet50_example.py
+```
+:::
+::::
+
+Then, build the corresponding docker images and push it to your choice of docker registry.
+
+::::{tab-set}
+:::{tab-item} Whisper
+```bash
+export IMG1={your-image-repo-and-name}
+docker build -t $IMG1 -f whisper.Dockerfile .
+docker push $IMG1
+```
+:::
+:::{tab-item} Resnet
+```bash
+export IMG2={your-image-repo-and-name}
+docker build -t $IMG2 -f resnet.Dockerfile .
+docker push $IMG2
+```
+:::
+::::
+
+Finally, you can specify the docker image within which you want to run each application in the `container` field of an application's runtime environment specification. The `container` field has three fields:
+- `image`: (Required) The image to run your application in.
+- `worker_path`: The absolute path to `default_worker.py` inside the container. This is necessary if the Ray installation directory is different than that of the host (where raylet is running). For instance, if you are running on bare metal, and using a standard Ray docker image as the base image for your application's, then the Ray installation directory on host will likely differ from that of the container. Note that if you are using a standard Ray docker image as the base image, then `default_worker.py` will always be found at `/home/ray/anaconda3/lib/{python-version}/site-packages/ray/_private/workers/default_worker.py`.
```

Comment:
Does this mean if I use a standard Ray docker image, I don't need to set this field?

If so, let's call it out more clearly that this is an advanced option that most people shouldn't need to think about. Also, how can users debug if this is set incorrectly? What will the symptom be?

I'd consider omitting this field here and have an advanced section below describing it & how to debug/set it properly.

Reply:
If the user is running the Ray cluster in a container already, my guess is this is not necessary because they'll most likely use the same base image, or at least use the same method to install ray in the images, so that the installation path is the same.

> Also, how can users debug if this is set incorrectly? What will the symptom be?

I'm actually not certain, I'll check on this and add to the doc.

> I'd consider omitting this field here and have an advanced section below describing it & how to debug/set it properly.

I'm not 100% sure the "common" case doesn't require setting this. In a typical production setting, perhaps, since we probably expect them to be running the Ray cluster in a container already. However this isn't in the production guide, nor are we advertising that as the common case, so I want users who don't fit into that "typical production setting" case to be able to follow along as well. I can shorten the description here, then add an additional advanced section below like you suggested describing it in more detail and adding debugging instructions?

Reply:
Shortening here and having a more detailed section w/ debugging sounds good to me

Reply:
Done!